README.md

```diff
@@ -34,544 +34,542 @@ gem 'hanami-validations'
+To better recognise the pattern, let’s get back to the example above. This time we will map the natural language rules with programming language rules.
+
+```
+         A name must be present  and be a string and its size must be included between 3 and 64.
+            👇            👇     👇       👇    👇      👇                           👇    👇
+validates(:name)      { present? &&       str?   &&      size?                        (3 ..  64) }
 
-Person.new.valid?                             # < true
-Person.new(name: '').valid?                   # < true
-Person.new(skills: '').valid?                 # < true
-Person.new(skills: ['ruby', 'hanami']).valid?  # < true
+```
+
+Now, I hope you’ll never format code like that, but in this case, that formatting serves well our purpose to show how Ruby’s  simplicity helps to define complex rules with no effort.
+
+From a high level perspective, we can tell that input data for `name` is _valid_ only if **all** the requirements are satisfied. That’s because we used `&&`, but we can use whatever logic we want. **Remember, they are just lambdas**.
 
-Person.new(skills: []).valid?                 # < false
-Person.new(keys: {}).valid?                   # < false
-Person.new(keys: {a: :b}, skills: []).valid?  # < false
+```ruby
+validates(:name) { true }
+validates(:name) { 1 == 1 }
+validates(:name) { 0 > 1 || 0 < 1 }
 ```
 
-If you want to _disable_ this behaviour, please, refer to [presence](https://github.com/hanami/validations#presence).
+These examples above are all acceptable, but not really useful. 
+We need something applicable to our real world projects.
+
+### Predicates
 
-### Validations
+To meet our needs, `Hanami::Validations` has an extensive collection of **built-in** predicates. **A predicate is the expression of a business requirement** (e.g. size greater than). The chain of several predicates determines if input data is valid or not.
 
-If you prefer Hanami::Validations to **only define validations**, but **not attributes**,
-you can use the following alternative syntax.
+We already met _presence_ and _size_, now let’s introduce the rest of them. They capture **common use cases with web forms**.
+
+#### Acceptance
+
+This predicate is useful to verify if a form check-box was checked by an user.
 
 ```ruby
-require 'hanami/validations'
+validates(:terms_of_service) { accepted? }
+```
 
-class Person
-  include Hanami::Validations
-  attr_accessor :name, :email
+_Truthy_ values are evaluated according to the following rules.
 
-  # Custom initializer
-  def initialize(attributes = {})
-    @name, @email = attributes.values_at(:name, :email)
-  end
+| value         | valid |
+|---------------|-------|
+| `false`       | ✗     |
+| `true`        | ✔︎     |
+| `nil`         | ✗     |
+| `""`          | ✗     |
+| `"0"`         | ✗     |
+| `"1"`         | ✔︎     |
+| other strings | ✗     |
+| `0`           | ✗     |
+| `1`           | ✔︎     |
+| other numbers | ✗     |
+| other objects | ✔︎     |
```

Comment:
GitHub renders these two quite differently. Perhaps you could use a different [Unicode checkmark](https://en.wikipedia.org/wiki/Check_mark)?
![screen shot 2016-04-20 at 09 44 11](https://cloud.githubusercontent.com/assets/632942/14680971/77c21b1c-06dc-11e6-81f2-eb4f95a94544.png)